cdap-ui/app/cdap/components/HeaderActions/index.js

```diff
@@ -52,6 +88,58 @@ export default class HeaderActions extends Component {
     });
   }
 
+  toggleNamespaceDropdown(){
+    this.setState({
+      namespaceOpen : !this.state.namespaceOpen
+    });
+  }
+
+  loadNamespacesFromStore (res, current) {
+
+    if(typeof res === 'undefined'){
+      return;
+    }
+
+    if(current){
+      current = current();
```

Comment:
If we pass the current as string then we shouldn't probably do this. In general it would be nice if we don't change the type of a function argument that is coming. Originally this is coming in as function but here we are modifying it as string or something else. It would be better if we avoid this